pennylane/ops/op_math/linear_combination.py

```diff
@@ -404,16 +361,25 @@ def compare(self, other):
                 pr2.simplify()
                 return pr1 == pr2
 
-            if isinstance(other, (LinearCombination, qml.ops.Hamiltonian)):
+            if isinstance(other, LinearCombination):
                 op1 = self.simplify()
                 op2 = other.simplify()
-                return op1._obs_data() == op2._obs_data()  # pylint: disable=protected-access
-
-            if isinstance(other, (Tensor, Observable)):
+                return qml.equal(op1, op2)
```

Comment:
This branch can probably be removed, since it's just the default behaviour written bellow.

Reply:
nice catch, thanks!